server/src/user/model.rs

```diff
@@ -127,6 +128,139 @@ impl User {
+                reason: "The requested User does not exist.".to_string()
+            })
+        }
+
+        sqlx::query!(
+                r#"update ad set author_id = ? where author_id = ?"#,
+                deleted_user_id,
+                payload.id,
+            )
+            .execute(&mut transaction)
+            .await?;
+
+
+        sqlx::query!(
+                r#"update blog_post set author_id = ? where author_id = ?"#,
+                deleted_user_id,
+                payload.id,
+            )
+            .execute(&mut transaction)
+            .await?;
+
+
+        sqlx::query!(
+                r#"update comment set author_id = ? where author_id = ?"#,
+                deleted_user_id,
+                payload.id,
+            )
+            .execute(&mut transaction)
+            .await?;
+
+        sqlx::query!(
+                r#"delete from comment_vote where user_id = ?"#,
+                payload.id,
+            )
+            .execute(&mut transaction)
+            .await?;
+
+        sqlx::query!(
+                r#"update entity_revision set author_id = ? where author_id = ?"#,
+                deleted_user_id,
+                payload.id,
+            )
+            .execute(&mut transaction)
+            .await?;
+
+        sqlx::query!(
+                r#"update event_log set actor_id = ? where actor_id = ?"#,
+                deleted_user_id,
+                payload.id,
+            )
+            .execute(&mut transaction)
+            .await?;
+
+        sqlx::query!(
+                r#"delete from flag where reporter_id = ?"#,
+                payload.id,
+            )
+            .execute(&mut transaction)
+            .await?;
+
+        sqlx::query!(
+                r#"delete from notification where user_id = ?"#,
+                payload.id,
+            )
+            .execute(&mut transaction)
+            .await?;
+
+        sqlx::query!(
+                r#"update page_revision set author_id = ? where author_id = ?"#,
+                deleted_user_id,
+                payload.id,
+            )
+            .execute(&mut transaction)
+            .await?;
+
+        sqlx::query!(
+                r#"delete from role_user where user_id = ?"#,
+                payload.id,
+            )
+            .execute(&mut transaction)
+            .await?;
+
+        sqlx::query!(
+                r#"delete from subscription where user_id = ?"#,
+                payload.id,
+            )
+            .execute(&mut transaction)
+            .await?;
+
+        sqlx::query!(
+                r#"delete from subscription where uuid_id = ?"#,
+                payload.id,
+            )
+            .execute(&mut transaction)
+            .await?;
+
+        sqlx::query!(
+                r#"delete from uuid where id = ? and discriminator = 'user'"#,
+                payload.id,
+            )
+            .execute(&mut transaction)
+            .await?;
```

Comment:
I suppose you want to discuss this part. Indeed, not that beautiful.
A workaround would be to use one for-loop for the updates and another for the deletes.
Something like
```
for table in ["subscription", "uuid" /*etc*/"]{ // not sure if it is going to work
sqlx::query!(
                r#"delete from ? where id = ? and discriminator = 'user'"#,
                table,
                payload.id
```

Reply:
> The problem with the for-loop is that the ids are called different in the different tables (id, user_id, author_id,...)
Good point.
In that case, it is good as it is.